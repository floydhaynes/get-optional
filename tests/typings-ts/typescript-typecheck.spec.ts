import {execFile} from 'child_process';
import {promisify} from 'util';
import * as path from 'path';

const checkTs = (configFilename: string) => new Promise((resolve, reject) => execFile(
	path.join(ROOT_DIR, 'node_modules/.bin/tsc.cmd'),
	`--noEmit --project ./${configFilename}.tsconfig.json`.split(' '),
	{
		cwd: path.join(__dirname, 'testbed'),
	},
	(error, stdout, stderr) => {
		if (error) {
			reject(stdout || stderr || error);
		} else {
			resolve(stdout);
		}
	},
));

describe('Correct usage from TS', () => {
	it('should typecheck', async () => {
		const result = await checkTs('should-pass');
		expect(result).toBe('');
	});
});

describe('Incorrect usage from TS', () => {
	it('should not typecheck', async () => {
		const result = checkTs('should-not-pass');
		await expect(result).rejects.toMatchSnapshot();
	});
});
