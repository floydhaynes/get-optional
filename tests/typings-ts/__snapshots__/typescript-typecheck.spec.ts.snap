// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incorrect usage from TS should not typecheck 1`] = `
"should-not-pass.ts(23,7): error TS2322: Type 'C | undefined' is not assignable to type 'C'.
  Type 'undefined' is not assignable to type 'C'.
"
`;
